client/src/services/api.js

```diff
@@ -1,9 +1,67 @@
 import axios from 'axios'
 
-let api = axios.create({ baseURL: '/api' })
+let api
 
-function exampleAPICall() {
-  return api.get('/').then((res) => res.data)
+function initApi(token) {
+  if (!token) {
+    //init api without token
+    api = axios.create({ baseURL: '/api' })
+  } else {
+    // if token exists then init authorized api
+    api = axios.create({
+      baseURL: '/api',
+      headers: { authorization: `Bearer ${token}` }
+    })
+  }
 }
 
-export { exampleAPICall }
+// collection of authentication methods
+const authApi = {
+  async signup(formData) {
+    let { data } = await api.post('/auth/signup', formData)
+
+    // get user info to return it back and authorize api
+    if (!data.error) {
+      initApi(data.token)
+      data = (await api.get('/auth/getLoggedUser')).data
+    }
+
+    return data
+  },
+
+  async signin(formData) {
+    let { data } = await api.post('/auth/signin', formData)
+
+    // get user info to return it back and authorize api
+    if (!data.error) {
+      initApi(data.token)
+      data = (await api.get('/auth/getLoggedUser')).data
+    }
+
+    return data
+  },
+
+  async getCachedUser() {
+    try {
+      let { data, status } = await api.get('/auth/getCachedToken', {
+        withCredentials: true
+      })
+
+      // status 204 means there is no cookie, meaning the user is not logged in
+      if (status === 204) {
+        return Promise.resolve({ user: null })
```

Comment:
Can you precise why status 204 means no cookie?

Reply:
status '204' means the http request's content, syntax and semantics were understood. In layman's terms, it's the server answering "Your request was good, but i have nothing to give you back", which is the way we thought is the best answer.
the alternative was a `res.json({ message: 'There is no cached token' })`, but response messages should be displayed to the user, and this one wasn't something we want to show.

Generally, all application layer logic should be answered with 2XX as far as I know, but what I'm sure of is that 4XX errors are reserved for http layer errors as opposed to application layer exceptions.